File: tests/expectations/item_types.c
#include <stdarg.h>
#include <stdbool.h>
#include <stdint.h>
#include <stdlib.h>

enum OnlyThisShouldBeGenerated
#if __STDC_VERSION__ >= 202311L
  : uint8_t
#endif // __STDC_VERSION__ >= 202311L
 {
  Foo,
  Bar,
};
#if __STDC_VERSION__ >= 202311L
typedef enum OnlyThisShouldBeGenerated OnlyThisShouldBeGenerated;
#else
typedef uint8_t OnlyThisShouldBeGenerated;
#endif // __STDC_VERSION__ >= 202311L

File: tests/expectations/item_types.java
import com.sun.jna.*;
import com.sun.jna.ptr.*;

enum BindingsSingleton {
  INSTANCE;
  final Bindings lib = Native.load("", Bindings.class);
}

interface Bindings extends Library {
  Bindings INSTANCE = BindingsSingleton.INSTANCE.lib;


  class OnlyThisShouldBeGenerated extends IntegerType {
    public OnlyThisShouldBeGenerated() {
      super(4, true);
    }

    public OnlyThisShouldBeGenerated(long value) {
      super(4, value, true);
    }

    public OnlyThisShouldBeGenerated(Pointer p) {
      this(p.getInt(0));
    }
    public static final OnlyThisShouldBeGenerated Foo = new OnlyThisShouldBeGenerated(1);
    public static final OnlyThisShouldBeGenerated Bar = new OnlyThisShouldBeGenerated(2);

  }

  class OnlyThisShouldBeGeneratedByReference extends ByReference {
    public OnlyThisShouldBeGeneratedByReference() {
      super(4);
    }

    public OnlyThisShouldBeGeneratedByReference(Pointer p) {
      super(4);
      setPointer(p);
    }

    public OnlyThisShouldBeGenerated getValue() {
      Pointer p = getPointer();
      return new OnlyThisShouldBeGenerated(p.getInt(0));
    }

    public void setValue(OnlyThisShouldBeGenerated value) {
      Pointer p = getPointer();
      p.setInt(0, value.intValue());
    }

  }


}